### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by ziolushka**
Have a couple of questions about the advisability of using the hardware_concurrency() function in this case. _The following code is written without._

### CODE REVIEW

File: Code_parser.cpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#include "Code_parser.h"
+
+void Code_parser::doParse(const string file_path) {
```
Comment (Stepbura) [inline Code_parser.cpp]: It is better to redefine the first function argument as a reference. Consider replacing 'const .. file_path' with 'const .. &file_path'.

File: Code_analyzer.h [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+	private:
+		int total_lines
+			, total_code_lines
+			, total_blank_lines
+			, total_comment_lines;
+		
+
+	public:
+		Code_analyzer() : total_lines(0), total_code_lines(0), total_blank_lines(0), total_comment_lines(0){};
+		void doAnalyze(const string);
```
Comment (Stepbura) [inline Code_analyzer.h]: The argument should probably be rendered as a constant reference.

File: Code_analyzer.h [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#pragma once
+#include "Code_parser.h"
+#include "File_finder.h"
+
+class Code_analyzer
+{
+	private:
+		int total_lines
+			, total_code_lines
```
Comment (Stepbura) [inline Code_analyzer.h]: Please create a new variable with type name and initialize it here.

File: Code_analyzer.h [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+			, total_blank_lines
+			, total_comment_lines;
+		
+
+	public:
+		Code_analyzer() : total_lines(0), total_code_lines(0), total_blank_lines(0), total_comment_lines(0){};
+		void doAnalyze(const string);
+		void saveResult(string) const;
```
Comment (Stepbura) [inline Code_analyzer.h]: The argument should probably be rendered as a constant reference.

File: Code_parser.cpp [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+	ifstream read_file(file_path);
```
Comment (Stepbura) [inline Code_parser.cpp]: I think it is better to verify if file exists first.

File: Code_parser.h [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+#include <regex>
+#include <stack>
+
+namespace fs = std::filesystem;
+using namespace std;
+
+class Code_parser
+{
+	private:
+		int blank_lines_count
```
Comment (Stepbura) [inline Code_parser.h]: It is better to declare the variable and then initializing it.

File: Code_parser.h [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+			, comment_lines_count
```
Comment (Stepbura) [inline Code_parser.h]: Please declare each variable and then initializing it.

File: Code_parser.h [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+			, code_lines_count
+			, physical_lines_count;
+
+	public:
+		Code_parser() :blank_lines_count(0), comment_lines_count(0), code_lines_count(0), physical_lines_count(0){};
+		void doParse(const string);
```
Comment (Stepbura) [inline Code_parser.h]: Use const reference as a parameter.

File: File_finder.h [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+using namespace std;
+
+class File_finder
+{
+	private:
+		string directory_path;
+		vector<string> directory_files_path;
+	public:
+		File_finder() : directory_path("") {};
+		File_finder(string dir_path) : directory_path(dir_path) {};
```
Comment (Stepbura) [inline File_finder.h]: Use string reference as a parameter.

File: File_finder.h [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+		int getFileCount();
```
Comment (Stepbura) [inline File_finder.h]: It makes sense to use const method

File: Code_analyzer.cpp [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+
+    ofstream output(directory_path + "Result.txt");
+    if (!output.is_open())
+        throw exception("can't open/create file");
+    output << result_info;
+
+    output.close();
+}
+void Code_analyzer::doAnalyze(string current_file) {
+
```
Comment (Stepbura) [inline Code_analyzer.cpp]: Why do not you use any thread synchronization method? You use mutual variables from different threads.

**Review by Stepbura — CHANGES_REQUESTED** [review]
Please take a look my comments.

File: Code_parser.cpp (hunk 1 above)
Comment (ziolushka) [reply Code_parser.cpp]: Resolved

File: Code_parser.h (hunk 8 above)
Comment (ziolushka) [reply Code_parser.h]: Resolved

File: File_finder.h (hunk 9 above)
Comment (ziolushka) [reply File_finder.h]: Resolved

File: File_finder.h (hunk 10 above)
Comment (ziolushka) [reply File_finder.h]: Resolved

File: Code_parser.cpp (hunk 5 above)
Comment (ziolushka) [reply Code_parser.cpp]: Resolved

File: Code_analyzer.cpp (hunk 11 above)
Comment (ziolushka) [reply Code_analyzer.cpp]: Resolved

File: File_finder.h (hunk 10 above)
Comment (ziolushka) [reply File_finder.h]: Resolved

File: Code_parser.h (hunk 7 above)
Comment (ziolushka) [reply Code_parser.h]: Resolved

File: Code_analyzer.h (hunk 4 above)
Comment (ziolushka) [reply Code_analyzer.h]: Resolved

File: Code_parser.h (hunk 6 above)
Comment (ziolushka) [reply Code_parser.h]: Resolved

File: Code_analyzer.h (hunk 3 above)
Comment (ziolushka) [reply Code_analyzer.h]: Resolved

File: Code_analyzer.h (hunk 2 above)
Comment (ziolushka) [reply Code_analyzer.h]: Resolved

**Pull request closed without merging**